samza-core/src/main/scala/org/apache/samza/container/TaskInstance.scala

```diff
@@ -181,22 +181,10 @@ class TaskInstance(
       trace("Processing incoming message envelope for taskName and SSP: %s, %s"
         format (taskName, incomingMessageSsp))
 
-      if (isAsyncTask) {
```

Comment:
Just double checking here too: Is this check no longer necessary?

Reply:
See my other comment - this is check is now redundant because `StreamTask` is no longer used internally but instead always gets wrapped by `AsyncStreamTaskAdapter`.